File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int total_sum=0;
        for(int i:cardPoints){
            total_sum+=i;
        }
        if(k==cardPoints.size()) return total_sum;
        int window_size=cardPoints.size()-k;
        int min_window=INT_MAX;
        int sum=0;
        for(int i=0;i<window_size;i++){
            sum+=cardPoints[i];
        }
        min_window=min(min_window,sum);
        for(int i=1;i<=k;i++){
            sum=sum-cardPoints[i-1]+cardPoints[window_size-1+i];
            min_window=min(min_window,sum);
        }
        return total_sum-min_window;
    }
};

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        # edge cases
        if(k==1):
            return max(cardPoints[0],cardPoints[-1])
        if(k==len(cardPoints)):
            return sum(cardPoints)

        n=len(cardPoints)
        window_size=n-k
        total_sum=sum(cardPoints)
        summy=0

        for i in range(0,n-k):
            summy+=cardPoints[i]
        maxi=total_sum-summy

        for i in range(1,k+1):
            summy=summy-cardPoints[i-1]+cardPoints[window_size+i-1]
            maxi=max(maxi,total_sum-summy)
        return maxi

